src/sentry/static/sentry/app/views/eventsV2/table/columnEditRow.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import SelectControl from 'app/components/forms/selectControl';
+import {OrganizationSummary, SelectValue} from 'app/types';
+import space from 'app/styles/space';
+
+import {
+  AGGREGATIONS,
+  FIELDS,
+  TRACING_FIELDS,
+  Field,
+  Aggregation,
+} from '../eventQueryParams';
+import {Column} from '../eventView';
+import {AGGREGATE_ALIASES} from '../data';
+
+type Props = {
+  organization: OrganizationSummary;
+  column: Column;
+  tagKeys: string[];
+  onChange: (column: Column) => void;
+};
+
+type State = {
+  field: string;
+  fields: SelectValue<string>[];
+  aggregation: Aggregation;
+  aggregations: string[];
+};
+
+class ColumnEditRow extends React.Component<Props, State> {
+  state = {
+    field: this.props.column.field,
+    fields: generateOptions(Object.keys(FIELDS).concat(this.props.tagKeys)),
+    aggregation: this.props.column.aggregation as Aggregation,
+    aggregations: filterAggregations(this.props.organization, this.props.column.field),
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.tagKeys !== this.props.tagKeys) {
+      this.syncFields();
+    }
+  }
+
+  handleFieldChange = ({value}) => {
+    this.setState((state: State, props: Props) => {
+      const newAggregates = filterAggregations(props.organization, value);
+      const newState = {...state, field: value, aggregations: newAggregates};
+
+      // If the new field makes the aggregation invalid, we should clear that state.
+      if (state.aggregation && !newAggregates.includes(state.aggregation)) {
+        // TODO(mark) Figure out why react-select isn't wiping the text value when this happens.
+        newState.aggregation = '';
+      }
+      return newState;
+    }, this.triggerChange);
+  };
+
+  handleFunctionChange = ({value}) => {
+    // TODO(mark) When we add improved tracing function support also clear
+    // the function parameter as necessary.
+    this.setState({aggregation: value}, this.triggerChange);
+  };
+
+  triggerChange() {
+    const {field, aggregation} = this.state;
+    this.props.onChange({
+      field,
+      aggregation,
+    });
+  }
+
+  syncFields() {
+    this.setState({
+      fields: generateOptions(Object.keys(FIELDS).concat(this.props.tagKeys)),
+    });
+  }
+
+  render() {
+    const {field, fields, aggregation, aggregations} = this.state;
+
+    return (
+      <RowContainer>
+        <SelectControl options={fields} value={field} onChange={this.handleFieldChange} />
+        <SelectControl
+          options={aggregations.map(item => ({label: item, value: item}))}
+          value={aggregation}
+          onChange={this.handleFunctionChange}
+        />
+      </RowContainer>
+    );
+  }
+}
+
+function generateOptions(values: string[]): SelectValue<string>[] {
+  return values.map(item => ({label: item, value: item}));
+}
+
+function filterAggregations(organization: OrganizationSummary, f?: Field): Aggregation[] {
```

Comment:
This introduces some duplication, but I'm optimizing for simple deletion in the future.